tensorflow/dtensor/mlir/utils/collective_lowering.cc

```diff
@@ -285,6 +285,113 @@ mlir::Operation* EmitTransposeOp(mlir::OpBuilder& builder,
                                                perm_op);
 }
 
+mlir::Operation* EmitCollectiveReduceScatter(
```

Comment:
We shall add a unit test for this new lowering. Although I noticed the MLIR rewrite unit tests are not OSSed, so there is no way for you to add one. Let me take a look why that's the case and get back to you.

Reply:
Thanks @rainwoodman, let me know when they are open sourced and I can add a test.

Reply:
The MLIR tests have been opened! An example to follow is probably this one: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/dtensor/mlir/tests/dtensor_all_gather.mlir

Reply:
Thank you! Will add this shortly

Reply:
Just added some tests, let me know if it looks good to you.